demo-app/src/main.rs

```diff
@@ -0,0 +1,253 @@
+use std::marker::PhantomData;
+
```

Comment:
Let's move the all the data structures `DemoTransaction`, `Demo` etc.. and trait impls to a separate file `types.rs` or `impl.rs`